core/src/main/java/ai/timefold/solver/core/impl/phase/AbstractPhase.java

```
@@ -88,18 +92,32 @@ public boolean isAssertShadowVariablesAreNotStaleAfterStep() {
 
     @Override
     public void solvingStarted(SolverScope<Solution_> solverScope) {
-        phaseTermination.solvingStarted(solverScope);
         phaseLifecycleSupport.fireSolvingStarted(solverScope);
     }
 
     @Override
     public void solvingEnded(SolverScope<Solution_> solverScope) {
-        phaseTermination.solvingEnded(solverScope);
         phaseLifecycleSupport.fireSolvingEnded(solverScope);
     }
 
     @Override
     public void phaseStarted(AbstractPhaseScope<Solution_> phaseScope) {
+        if (!solver.isTerminationSameAsSolverTermination(phaseTermination)) {
+            // Only fail if the user put the inapplicable termination on the phase, not on the solver.
+            // On the solver level, inapplicable phase terminations are skipped.
+            // Otherwise you would only be able to configure a global phase-level termination on the solver
+            // if it was applicable to all phases.
+            var unsupportedTerminationList = phaseTermination instanceof UniversalTermination<Solution_> universalTermination
+                    ? universalTermination.getPhasesTerminationsInapplicableTo(phaseScope)
+                    : Collections.emptyList();
+            if (!unsupportedTerminationList.isEmpty()) {
```

Can we fail fast when the solver is being built? This will fail during solving.